lib/oj/bag.rb

```diff
@@ -37,9 +36,9 @@ def respond_to?(m)
     # @raise [NoMethodError] if the instance variable is not defined.
     def method_missing(m, *args, &block)
       raise ArgumentError.new("wrong number of arguments (#{args.size} for 0) to method #{m}") unless args.nil? or args.empty?
-      at_m = ('@' + m.to_s).to_sym
-      raise NoMethodError.new("undefined method #{m}", m) unless instance_variable_defined?(at_m)
-      instance_variable_get(at_m)
+      result = instance_variable_get(:"@#{m}")
+      raise NoMethodError.new("undefined method #{m}", m) unless result
+      result
     end
```

Comment:
OK. I'd make the change with `instance_variable_defined?`. I didn't realize this use case

To another one problem. See this benchmark:

```
ruby 2.2.2p95 (2015-04-13 revision 50295) [x86_64-darwin14]

Calculating -------------------------------------
                fast    99.381k i/100ms
                slow    97.281k i/100ms
-------------------------------------------------
                fast      3.020M (±12.1%) i/s -     14.907M
                slow      2.811M (±10.0%) i/s -     13.911M

Comparison:
                fast:  3019754.8 i/s
                slow:  2811132.8 i/s - 1.07x slower
```

Code:
```ruby
require 'benchmark/ips'

variable = 'count'

Benchmark.ips do |x|
  x.report('fast') { :"@#{variable}"}
  x.report('slow') { ('@' + variable).to_sym }
  x.compare!
end
```

Here are some useful tips how some approaches of same things are slow/fast: https://github.com/JuanitoFatas/fast-ruby

Maybe are there some useful tips how to improve C code according to it (I don't understand C code so I may be totally wrong). It depends what equivalents in C of that methods called.